network/containerizer/bridgepolicy.go

```diff
@@ -0,0 +1,303 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package containerizer
+
+import (
+	"fmt"
+
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+	"github.com/juju/utils/set"
+
+	"github.com/juju/juju/network"
+	// Used for some constants and things like LinkLayerDevice[Args]
+	"github.com/juju/juju/state"
+)
+
+var logger = loggo.GetLogger("juju.network.containerizer")
+
+// BridgePolicy defines functionality that helps us create and define bridges
+// for guests inside of a host machine, along with the creation of network
+// devices on those bridges for the containers to use.
+// Ideally BridgePolicy would be defined outside of the 'state' package as it
+// doesn't deal directly with DB content, but not quite enough of State is exposed
+type BridgePolicy struct {
+}
+
+// Machine describes either a host machine, or a container machine. Either way
+// *state.Machine should fulfill the interface in real code.
+type Machine interface {
+	Id() string
+	AllSpaces() (set.Strings, error)
+	LinkLayerDevicesForSpaces([]string) (map[string][]*state.LinkLayerDevice, error)
```

Comment:
The return could just be `[]*state.LinkLayerDevice, err`. From a callers perspective order may be important. And the result can be indexed via its input []string.